webrtc.html

```diff
@@ -656,30 +656,34 @@ <h2>Dictionary <dfn>RTCOfferAnswerOptions</dfn>
         </div>
         <div>
           <pre class="idl">dictionary RTCOfferOptions : RTCOfferAnswerOptions {
-             boolean iceRestart = false;
+             boolean iceRestart;
 };</pre>
           <section>
             <h2>Dictionary <dfn>RTCOfferOptions</dfn> Members</h2>
             <dl data-link-for="RTCOfferOptions" data-dfn-for="RTCOfferOptions"
             class="dictionary-members">
               <dt><dfn data-idl><code>iceRestart</code></dfn> of type <span class=
-              "idlMemberType">boolean</span>, defaulting to
-              <code>false</code></dt>
+              "idlMemberType">boolean</span></dt>
               <dd>
-                <p>When the value of this dictionary member is true, the
-                generated description will have ICE credentials that are
+                <p>When the value of this dictionary member is <code>true</code>,
+                the generated description will have ICE credentials that are
                 different from the current credentials (as visible in the
                 <code><a data-link-for=
                 "RTCPeerConnection">localDescription</a></code> attribute's
                 SDP). Applying the generated description will restart ICE, as
                 described in section 9.1.1.1 of [[!ICE]].</p>
-                <p>When the value of this dictionary member is false, and the
-                <code><a data-link-for=
+                <p>When the value of this dictionary member is <code>false</code>,
+                and the <code><a data-link-for=
```

Comment:
Do we want "iceRestart = false" to ignore restartIce()? It makes sense in terms of meaning, and onnegotiationneedede would still fire when getting back to stable, but is there a reason not to treat "= false" the same as missing? It adds complexity and means there are effectively three values of iceRestart, not two.

I'm not convinced that we should change this, but I think it's worth discussing before merging this.

Reply:
(I'm also not convinced we shouldn't change it ;) )

Reply:
I still need to change the WPT tests. With the latest change where we no longer clear the internal slot always in SRD(answer), the effect of `{iceRestart: false}` will merely be to produce one non-restarting offer. It won't terminally cancel the restart the way it used to (meaning the engine will still fire `negotiationneeded` until the restart job is done.

That seems reasonable and benign to me. But at the same time it now seems kinda useless. There's really no way to cancel the restart now. Perhaps that is fine, but then perhaps we should also get it over with ASAP, and not leave these loopholes where one can drag it out.

We could also change it to the way you mention. It just seemed odd to me to restart ICE in an explicit`createOffer({iceRestart: false})` call. But we can do that.

Reply:
For what it's worth, it would be easier to implement iceRestart: false as the same as undefined, because then we don't have to introduce a third value in all the layers in chromium and webrtc.org. There's also less to test. And as you say, you can't abort a restart, so why would you generate an offer that doesn't restart - you're just wasting time.

In the future, setLocalDescription() will generate the offer for you anyway, so I don't see the need to maintain this level of control.

Reply:
If you did restartIce() + createOffer({iceRestart: false}) it was likely not intentional.

Reply:
Fixed. PTAL

Reply:
Looks good, just make sure to revert the dictionary change, i.e. so that false is still the default value for it!